source/game/system/RaceConfig.cc

```diff
@@ -15,32 +13,52 @@ void RaceConfig::init() {
 
 /// @addr{0x805302C4}
 /// @details Normally we copy the menu scenario into the race scenario.
-/// There's no menu scenario in Kinoko, so instead we initialize values here.
-/// For now, we want to initialize solely based off the ghost file.
+/// However, Kinoko doesn't support menus, so we use a callback.
 void RaceConfig::initRace() {
     m_raceScenario.playerCount = 1;
 
-    size_t size;
-    const auto *testDirector = Host::KSystem::Instance().testDirector();
-    u8 *rkg = Abstract::File::Load(testDirector->testCase().rkgPath.data(), size);
-    m_ghost = rkg;
-    delete[] rkg;
+    if (s_onInitCallback) {
+        s_onInitCallback(this, s_onInitCallbackArg);
+    }
+
+    initControllers();
+}
+
+/// @addr{0x8052F4E8}
+/// @brief Initializes the controllers.
+/// @details This is normally scoped within RaceConfig::Scenario, but Kinoko doesn't support menus.
+void RaceConfig::initControllers() {
+    switch (m_raceScenario.players[0].type) {
+    case Player::Type::Ghost:
+        initGhost();
+        break;
+    case Player::Type::Local:
+        KPadDirector::Instance()->setHostPad(m_raceScenario.players[0].driftIsAuto);
+        break;
+    default:
+        PANIC("Players must be either local or ghost!");
+        break;
+    }
+}
+
+/// @addr{0x8052EEF0}
+/// @brief Initializes the ghost.
+/// @details This is normally scoped within RaceConfig::Scenario, but Kinoko doesn't support menus.
+void RaceConfig::initGhost() {
     GhostFile ghost(m_ghost);
 
     m_raceScenario.course = ghost.course();
     Player &player = m_raceScenario.players[0];
     player.character = ghost.character();
     player.vehicle = ghost.vehicle();
-    player.type = Player::Type::Ghost;
+    player.driftIsAuto = ghost.driftIsAuto();
 
-    initControllers(ghost);
+    KPadDirector::Instance()->setGhostPad(ghost.inputs(), ghost.driftIsAuto());
 }
 
-/// @addr{0x8052F4E8}
-/// @details This is normally scoped within RaceConfig::Scenario, but since Kinoko doesn't support
-/// menus, we simplify and just initialize the controller here.
-void RaceConfig::initControllers(const GhostFile &ghost) {
-    KPadDirector::Instance()->setGhostPad(ghost.inputs(), ghost.driftIsAuto());
+void RaceConfig::RegisterInitCallback(InitCallback callback, void *arg) {
```

Comment:
```suggestion
void RaceConfig::RegisterInitCallback(const InitCallback &callback, void *arg) {
```
I think this is legal

Reply:
Yeah, it is. I had tried this but gave up before I realized `callback` *must* be const.